src/NGen.cpp

```diff
@@ -53,12 +68,166 @@ pdm03_struct get_et_params() {
     return pdm_et_data;
 }
 
+#ifdef NGEN_MPI_ACTIVE
+
+/**
+ * Perform an "AND" type sync across all ranks to synchronize whether all are in a 'success' or 'ready' state.
+ *
+ * Function should be called at the same time by all ranks to sync on some 'success' or 'ready' state for all the ranks.
+ * It handles the required MPI communication and the processing of received message content.
+ *
+ * Function accepts a value for some boolean status property.  This initial value is applicable only to the local MPI
+ * rank, with all MPI ranks having the status property and each having its own independent value.  The function has all
+ * ranks (except ``0``) send their local status to rank ``0``.  Rank ``0`` then applies a Boolean AND to the statuses
+ * (including its own) to produce a global status value.  The global status is then broadcast back to the other ranks.
+ * Finally, the value indicated by this global status is returned.
+ *
+ * @param status The initial individual state for the current MPI rank.
+ * @param taskDesc A description of the related task, used by rank 0 to print a message when included for any rank that
+ *                 is not in the success/ready state.
+ * @return Whether all ranks coordinating status had a success/ready status value.
+ */
+bool mpiSyncStatusAnd(bool status, const std::string &taskDesc) {
```

Comment:
I would like to this function in a different compilation unit and included/used in the driver if needed.

Reply:
Moved to dedicated file.